tests/lib/tools/suite/fs-state/task.yaml

```diff
@@ -0,0 +1,57 @@
+summary: smoke test for the fs-state tool
+
+execute: |
+    # Check help
+    "$TESTSTOOLS"/fs-state -h | MATCH "usage: fs-state wait-for-file"
+    "$TESTSTOOLS"/fs-state --help | MATCH "usage: fs-state wait-for-file"
+    "$TESTSTOOLS"/fs-state | MATCH "usage: fs-state wait-for-file"
+
+    # Check non existing directory
+    "$TESTSTOOLS"/fs-state mock-dir mytestdir
+    test -d mytestdir
+    test -f mytestdir.fake
+    not test -d mytestdir.back
+
+    "$TESTSTOOLS"/fs-state restore-dir mytestdir
+    not test -d mytestdir
+
+    # Check existing directory
+    mkdir mytestdir
+    "$TESTSTOOLS"/fs-state mock-dir mytestdir
+    test -d mytestdir
+    test -f mytestdir.fake
+    test -d mytestdir.back
+    "$TESTSTOOLS"/fs-state restore-dir mytestdir
+    test -d mytestdir
+    rm -r mytestdir
+
+    # Restoring a dir which is not mocked is not allowed
+    mkdir mytestdir
+    not "$TESTSTOOLS"/fs-state restore-dir mytestdir
```

Comment:
Could you also test the error message please. It will be useful in case there's a logic mistake in a test they are writing or debugging.

Reply:
done